src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -116,6 +116,149 @@ inline int io_getevents(aio_context_t ctx, long min_nr, long max_nr,
 
 namespace xocl {
 
+/*
+ * queue_cb: control block for a qdma queue
+ */
+class queue_cb {
+public:
+    friend class shim;
+
+    queue_cb(struct xocl_qdma_ioc_create_queue *qinfo);
+    ~queue_cb();
+    int set_option(int type, uint32_t val);
+    // get aio completion event of the queue
+    int queue_poll_completion(int min_compl, int max_compl,
+                    struct xclReqCompletion *comps, int* actual, int timeout);
+    // submit the read/write i/o to the queue
+    ssize_t queue_submit_io(xclQueueRequest *wr, aio_context_t *mAioCtx);
+
+private:
+    uint64_t qhndl;		/* queue handle */
+    bool h2c; 			/* queue is for H2C direction */
+    bool qAioEn;		/* per queue aio is enabled */
+    aio_context_t qAioCtx;	/* per queue aio context */
+};
+
+queue_cb::~queue_cb()
```

Comment:
This is a general comment for all the member functions of queue_cb.  Since struct queue_cb is scoped to this compilation unit, it is better to define all this functions with the declaration in scope of the class definition.  There is nothing gained by separating the function definition from the struct member function declarations.